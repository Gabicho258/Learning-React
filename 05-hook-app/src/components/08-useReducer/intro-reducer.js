
const initialState = [{
    id: 1,
    todo: 'Comprar microfono',
    done: false
}]

const todoReducer = (state = initialState, action) =>{
    if (action?.type === 'agregar') {
        return [...state, action.payload];
    }
    return state;
}

let todos = todoReducer(); 

const newTodo = {
    id: 2,
    todo: 'Comprar keyboard',
    done: true

}

const action = {
    type: 'agregar',
    payload : newTodo

}

todos = todoReducer(todos, action);



console.log(todos);









