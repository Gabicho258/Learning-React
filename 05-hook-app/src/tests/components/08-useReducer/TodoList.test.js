import React from "react";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";
import { TodoList } from "../../../components/08-useReducer/TodoList";
describe("pruebas en <TodoList/>", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();
  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      //   index={1}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de tener dos <TodoListItem/>", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(
      expect.any(Function)
    );
  });
});
