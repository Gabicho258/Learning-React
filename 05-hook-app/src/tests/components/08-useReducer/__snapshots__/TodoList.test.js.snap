// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoList/> debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    todo={
      Object {
        "desc": "Aprender React",
        "done": true,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
